### ISSUE TITLE
Non-ASCII Name Problem in downloading generated QR Images

### ISSUE THREAD

**Issue by hamidpeywasti** [terminal42/contao-shortlink#27]
Hi,
There's an "Internal Server Error" in generating QR when the "Name" is not ASCII (in my case Persian - Unicode).

Error massage:
> The filename fallback must only contain ASCII characters.

Isn't it better if this feature works even with non-ASCII names? or mybe convert name to ASCII one before downloading?

I know that it causes in symfony/http-foundation.

'![image](https://github.com/user-attachments/assets/30c8e48a-01bf-4c37-a9a2-886ad97359ad)

**Comment by fritzmg** [comment]
The return value of `$shortlink->getName()` needs to be normalized here:

https://github.com/terminal42/contao-shortlink/blob/77710b24fa21eb76305af7b3f8befcc2b3d9a6a9/src/Controller/QrCodeController.php#L63-L82

Or alternatively provide the 3rd argument (`$filenameFallback`) for `HeaderUtils::makeDisposition()` - with a normalized version of the shortlink's name.

**Comment by aschempp** [comment]
any idea how to normalize this? Would it need the Transliterator? @hamidpeywasti what file name would you expect to have for the image in this case?

**Comment by fritzmg** [comment]
I'd just use the slug generator.

```php
$filename = ($this->slug->generate($shortlink->getName()) ?: 'qrcode').'.'.$format; 
// …
$response->headers->set('Content-Disposition', HeaderUtils::makeDisposition( 
    HeaderUtils::DISPOSITION_ATTACHMENT, 
    $filename, 
));
```

**Comment by aschempp** [comment]
Fixed in 3d61ddc4f49c8bf1335103c177b71760d6e2d741
